### PR TITLE
Python 3.12 patch

### PR DESCRIPTION
**Pull request by Thrameos**
This should fix Python 3.12 though not without serious wrinkles.   The new PyLong mutates a field in such a way that it will break anything that wants to add memory to the tail of the object to save a few bytes.   I though that given that we were following the exact mechanism that Python requires for its own structures that we would be safe.  The mutation is entirely in private parts of Python so I can't just use the private symbols so we track with any changes.   It doesn't just break our code but also the dictoffset and weakoffset fields in the Python base class, so I can't see that it will remain in the current state.  Thus this is at best a version that will work until the repair to the memory model is done, and then we likely have to rework another time.
   
Changes include:
- switch to mallinfo2 to avoid deprecation warning.
- change to PyLong model
- A shift in PyBuffer models that came in 3.9 but didn't burn us until now.
- Fix for wstr which may not be well covered 
- Disable part of the doc system until I can figure out what Java 17 did
- Update the tests to avoid deprecation warnings

### CODE REVIEW

File: native/python/pyjp_module.cpp [hunk 1]
```diff
@@ -169,11 +169,21 @@ PyObject* PyJP_GetAttrDescriptor(PyTypeObject *type, PyObject *attr_name)
 	if (type->tp_mro == NULL)
 		return NULL;  // GCOVR_EXCL_LINE
 
+	// Grab the mro
 	PyObject *mro = type->tp_mro;
+
+	// mco should be a tuple
```
Comment (marscher) [inline native/python/pyjp_module.cpp:175]: tiny typo: MRO

**Review by marscher — APPROVED** [review]
Thank you so much @Thrameos. Could you please merge latest master to enable testing for Python 3.12?
I think we should add a nightly version of Python to our matrix (if that'd be installable easily) in order to see such upcoming problems earlier. However this could also introduce lots of noise during the development of a new Python minor (which is apprently a major in semantic versioning related to the amount of breaking changes).

**Comment by pelson** [comment]
Looks like it may be this change which started a Python 3.7 nightly test to fail. Example in https://gitlab.cern.ch/scripting-tools/pyjapc/-/jobs/34505253.

I'm pinned back right now, but can investigate later next week if needed.

**Comment by Thrameos** [comment]
Okay I can roll that back.   I will have to investigate which systems have mallinfo vs mallinfo2.

**Pull request merged**